jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -1434,12 +1434,19 @@ else if (_path != null)
             {
                 // The RFC requires this to be canonical before decoding, but this can leave dot segments and dot dot segments
                 // which are not canonicalized and could be used in an attempt to bypass security checks.
-                _canonicalPath = URIUtil.canonicalPath(_path);
+                _canonicalPath = URIUtil.canonicalPath(_path, this::onBadUtf8);
                 if (_canonicalPath == null)
                     throw new IllegalArgumentException("Bad URI");
             }
         }
 
+        private RuntimeException onBadUtf8()
```

Comment:
Why is this returning anything at all?
It always returns null below.

Reply:
Because it is being passed in as a `Supplier<X extends Throwable> onBadUtf8` to `URIUtil.canonicalPath`.

In this case we don't want to throw, so we supply null, but we remember that there was a violation